### PR TITLE
[HOPSWORKS-809]  Fix exception handling bug

### PR DESCRIPTION
**Pull request by kouzant**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-809

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java [hunk 1]
```diff
@@ -305,29 +305,26 @@ public Response startNotebookServer(JupyterSettings jupyterSettings, @Context Ht
...
-            dfso, certificateMaterializer, settings, dto.getCertificatesDir());
+        HopsUtils.materializeCertificatesForUserCustomDir(project.getName(), user.getUsername(),
+            settings.getHdfsTmpCertDir(), dfso, certificateMaterializer, settings, dto.getCertificatesDir());
         // When Livy launches a job it will look in the standard directory for the certificates
         // We materialize them twice but most probably other operations will need them too, so it is OK
         // Remember to remove both when stopping Jupyter server or an exception is thrown
         certificateMaterializer.materializeCertificatesLocal(user.getUsername(), project.getName());
-      } catch (IOException | ServiceException ex) {
+      } catch (ServiceException ex) {
         LOGGER.log(Level.SEVERE, null, ex);
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java]: @kouzant not needed, it's logged by the throwablemapper

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java [hunk 2]
```diff
@@ -305,29 +305,26 @@ public Response startNotebookServer(JupyterSettings jupyterSettings, @Context Ht
...
-            settings.getHdfsTmpCertDir(),
-            certificateMaterializer, dto.getCertificatesDir(), settings);
-        } else {
-          throw new HopsSecurityException(RESTCodes.SecurityErrorCode.CERT_LOCATION_UNDEFINED, Level.SEVERE);
-        }
         throw new ServiceException(RESTCodes.ServiceErrorCode.JUPYTER_START_ERROR, Level.SEVERE, ex.getMessage(),
-          null, ex);
+            null, ex);
+      } catch (IOException ex) {
+        LOGGER.log(Level.SEVERE, "Jupyter certificate materialization error", ex);
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java]: @kouzant not needed, it's logged by the throwablemapper

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java (hunk 1 above)
Comment (kouzant) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java]: I removed logging of exception

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java (hunk 2 above)
Comment (kouzant) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java]: I removed logging of exception

**Pull request merged**